internal_workflow_testsuite.go

```diff
@@ -343,8 +341,10 @@ func (env *testWorkflowEnvironmentImpl) executeActivity(
 		input,
 	)
 
-	// ensure activityFn is registered to defaultTestTaskList
-	env.testSuite.RegisterActivity(activityFn)
+	if _, ok := getHostEnvironment().getActivityFn(fnName); !ok {
```

Comment:
Made this change in my PR itself. Removed the Register* methods. For a self contained test, you can still create WorkflowTestSuite instance with a new host environment and register on that environment.